### PR TITLE
[STORM-1204] Fixing attempt to access to directory before authorizing user

### PR DESCRIPTION
**Pull request by kishorvpatil**
Fixing dumps endpoints, if URL request has no file/directory it return 500 instead of page-not-found.

### CODE REVIEW

File: storm-core/src/clj/backtype/storm/daemon/logviewer.clj [hunk 1]
```diff
@@ -967,15 +978,30 @@
...
+                           topo-id
+                           file-path-separator
+                           port))
+           file (File. (str log-root
+                            file-path-separator
+                            topo-id
+                            file-path-separator
+                            port
+                            file-path-separator
+                            filename))]
```
Comment (redsanket) [inline storm-core/src/clj/backtype/storm/daemon/logviewer.clj]: Can we convert this to a utility method, it will be useful

**Comment by redsanket** [comment]
minor suggestion, +1 NB

**Comment by revans2** [comment]
+1

**Comment by kishorvpatil** [comment]
@redsanket We are doing this only are couple of places.. no making it a utility may not be much useful.

**Comment by zhuoliu** [comment]
+1

**Pull request merged**